src/js/events.js

```diff
@@ -268,163 +255,159 @@ jQuery( document ).ready( function() {
-                                    current_approximations.convergent_indicies,
-                                    current_approximations.ratios,
-                                    current_approximations.numerator_limits,
-                                    current_approximations.denominator_limits,
-                                    current_approximations.ratio_limits);
-        modify_update_approximations();
-  });
-  
-  // recalculate approximations when scale degree changes
-  jQuery( "#input_scale_degree").change( function() {
-  jQuery( "#txt_tuning_data" ).val( jQuery( "#txt_tuning_data" ).val().trim() );
+    current_approximations.convergent_indicies = [];
+    current_approximations.numerators = [];
+    current_approximations.denominators = [];
+    current_approximations.ratios = [];
+    current_approximations.numerator_limits = [];
+    current_approximations.denominator_limits = [];
+    current_approximations.ratio_limits = [];
+
+    get_rational_approximations(
+	  interval, 
+	  current_approximations.numerators, 
+	  current_approximations.denominators, 
+	  999999,
+      current_approximations.convergent_indicies,
+      current_approximations.ratios,
+      current_approximations.numerator_limits,
+      current_approximations.denominator_limits,
+      current_approximations.ratio_limits
+	);
 
-  if ( isEmpty(jQuery( "#txt_tuning_data" ).val()) ) {
-    alert( "No tuning data to modify." );
-    return false;
-   }
+    modify_update_approximations();
+  } );
 
+  // recalculate approximations when scale degree changes
+  jQuery( "#input_scale_degree").change( function() {
+    jQuery( "#txt_tuning_data" ).val( jQuery( "#txt_tuning_data" ).val().trim() );
     var index = parseInt( jQuery( '#input_scale_degree' ).val() ) - 1;
-  var lines = document.getElementById("txt_tuning_data").value.split(newlineTest);
-  jQuery ( "#input_interval_to_approximate" ).val(lines[index]);
-  jQuery ( "#input_interval_to_approximate" ).trigger("change");
-
-  });
+    var lines = document.getElementById("txt_tuning_data").value.split(newlineTest);
+    jQuery ( "#input_interval_to_approximate" ).val(lines[index]);
+    jQuery ( "#input_interval_to_approximate" ).trigger("change");
+  } );
 
   // refilter approximations when error amount changes
   jQuery( "#input_min_error" ).change( function() {
-      modify_update_approximations();
-  })
-                                                   
+    modify_update_approximations();
+  } );
+
    // refilter approximations when error amount changes
   jQuery( "#input_max_error" ).change( function() {
-      modify_update_approximations();
-  })
-                                                   
+    modify_update_approximations();
+  } );
+
   // refilter approximations when "show semiconvergents" changes
   jQuery( "#input_show_convergents" ).change( function() {
-      modify_update_approximations();
-  })
+    modify_update_approximations();
+  } );
```

Comment:
for future refactors: since all 3 change handlers do the same, the selectors can be grouped together into a single selector separated with commas. Also, `modify_update_approximations` need not to be wrapped into an anonymous function, it can be direcly passed to `.change()`

Reply:
Trying to use

`  jQuery( "#input_min_error, #input_max_error, #input_show_convergents" ).change( modify_update_approximations() );
`

But it doesn't seem to be working. Hmm